internal/k8s/deployments_test.go

```diff
@@ -557,3 +559,195 @@ func TestComputeConnectionSpecHash(t *testing.T) {
+				assert.Equal(t, expectedValue, actualValue, "Environment variable %s should have correct value", expectedKey)
+			}
+
+			// Verify unexpected environment variables are not present
+			for _, unexpectedKey := range tt.unexpectedEnvVars {
+				_, exists := envMap[unexpectedKey]
+				assert.False(t, exists, "Environment variable %s should not be present", unexpectedKey)
+			}
+
+			// For mTLS case, verify volume mounts and volumes are configured
+			if tt.connection.MutualTLSSecret != "" {
+				assert.Len(t, container.VolumeMounts, 1)
+				assert.Equal(t, "temporal-tls", container.VolumeMounts[0].Name)
+				assert.Equal(t, "/etc/temporal/tls", container.VolumeMounts[0].MountPath)
+
+				assert.Len(t, deployment.Spec.Template.Spec.Volumes, 1)
+				assert.Equal(t, "temporal-tls", deployment.Spec.Template.Spec.Volumes[0].Name)
+				assert.Equal(t, tt.connection.MutualTLSSecret, deployment.Spec.Template.Spec.Volumes[0].VolumeSource.Secret.SecretName)
+			}
+		})
+	}
+}
+
+func TestNewDeploymentWithOwnerRef_EnvConfigSDKCompatibility(t *testing.T) {
+	// Test that the environment variables injected by the controller
+	// can be parsed by the official Temporal SDK envconfig package
+
+	tests := map[string]struct {
+		connection      temporaliov1alpha1.TemporalConnectionSpec
+		expectedAddress string
+		expectTLS       bool
+	}{
+		"without TLS": {
+			connection: temporaliov1alpha1.TemporalConnectionSpec{
+				HostPort: "test.temporal.example:9999",
+			},
+			expectedAddress: "test.temporal.example:9999",
+			expectTLS:       false,
+		},
+		"with TLS": {
+			connection: temporaliov1alpha1.TemporalConnectionSpec{
+				HostPort:        "mtls.temporal.example:8888",
+				MutualTLSSecret: "test-tls-secret",
+			},
+			expectedAddress: "mtls.temporal.example:8888",
+			expectTLS:       true,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			spec := &temporaliov1alpha1.TemporalWorkerDeploymentSpec{
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "worker",
+								Image: "temporal/worker:latest",
+							},
+						},
+					},
+				},
+				WorkerOptions: temporaliov1alpha1.WorkerOptions{
+					TemporalNamespace: "test-namespace",
+				},
+			}
+
+			deployment := k8s.NewDeploymentWithOwnerRef(
+				&metav1.TypeMeta{},
+				&metav1.ObjectMeta{Name: "test", Namespace: "default"},
+				spec,
+				"test-deployment",
+				"test-build-id",
+				tt.connection,
+			)
+
+			// Extract environment variables from the deployment
+			container := deployment.Spec.Template.Spec.Containers[0]
+
+			// Set environment variables using t.Setenv() to simulate the runtime environment
+			for _, env := range container.Env {
+				t.Setenv(env.Name, env.Value)
+			}
+
+			// For TLS tests, verify environment variables are set but skip actual client loading
+			// because creating valid test certificates is complex
+			if tt.expectTLS {
```

Comment:
It's not that hard to create a test cert; try just creating empty files with the right names first, and if that doesn't work then generate a self-signed cert.